scipy/stats/_distribution_infrastructure.py

```diff
@@ -2461,9 +2461,7 @@ def _cdf2_subtraction_safe(self, x, y, **params):
         cdf_max = np.maximum(cdf_x, cdf_y)
         ccdf_max = np.maximum(ccdf_x, ccdf_y)
         spacing = np.spacing(np.where(i, ccdf_max, cdf_max))
-
-        with np.errstate(divide='ignore'):
-            mask = tol < abs(spacing/out)
+        mask = abs(tol * out) < spacing
```

Comment:
This is an improvement because we avoid division by `out`, which can be zero. I think it's also clearer: `spacing` is the absolute error estimate due to limited precision of the values `ccdf_max` and `cdf_max`; `tol * out` is the absolute error tolerance.

Note that `out` can be negative because it's the difference between probabilities, but `spacing` cannot be because `ccdf_max` and `cdf_max` are always between 0 and 1.